R/aggregate_by_matrix.R

```diff
@@ -0,0 +1,37 @@
+#' @rdname aggregate
+#'
+#' @export
+colMeansMat <- function(x, MAT)
+  colSumsMat(x, MAT)/colSums(MAT)
+
+#' @rdname aggregate
+#'
+#' @export
+colSumsMat <- function(x, MAT)
+  t(MAT) %*% x
+
+
+##' @param MAT An adjacency matrix that defines what features with
+##'     `nrow(MAT) == nrow(x)`
+##'
+##' @return [aggregate_by_matrix()] returns a `matrix` (or `Matrix`)
+##'     of dimensions `ncol(MAT)` and `ncol(x), with `dimnames` equal
+##'     to `colnames(x)` and `rownames(MAT)`.
+##'
+##' @rdname aggregate
+##'
+##' @export
+aggregate_by_matrix <- function(x, MAT, FUN, ...) {
+    if (!inherits(x, "matrix"))
+        stop("'x' must be a matrix.")
+    if (!inherits(MAT, "Matrix") & !inherits(MAT, "matrix"))
```

Comment:
I'm often unsure when to use `inherits` or `is`. Isn't `Matrix` a *S4* class and should be tested with `is`?

Reply:
There is indeed a small difference:

> The analogue of ‘inherits’ for formal classes is is.  The two
     functions behave consistently with one exception: S4 classes can
     have conditional inheritance, with an explicit test.  In this
     case, is will test the condition, but ‘inherits’ ignores all
     conditional superclasses.

And `is()` is more appropriate here. Will change